### PR TITLE
tiflash: support session and global variable for FastScan instead of ddl

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

- Add the usage of session/global variable 'tiflash_fastscan'.
- delete the related description about the ddl 'alter table set tiflash mode xx'

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/11151
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [x] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-use-fastscan.md [hunk 1]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
+The variable `tiflash_fastscan` supports session-level and global-level modifications. If you need to enable the FastScan feature in the current session, you can do so with the following statement:
+
+```
+set session tiflash_fastscan=ON;
 ```
 
-Once enabled, subsequent queries of this table in TiFlash will use the function of FastScan.
+You can also set `tiflash_fastscan` at the global level to enable new values for the default session and global variable `tiflash_fastscan` in the new session after setting.
```
Comment (hongyunyan) [inline develop/dev-guide-use-fastscan.md:48]: This sentence seems a little difficult to understand easily, could you please change it to more understandable?

**Review by hongyunyan — APPROVED** [review]
Rest is LGTM.

File: develop/dev-guide-use-fastscan.md (hunk 1 above)
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:48]:
```suggestion
You can also set `tiflash_fastscan` at the global level. The new setting then takes effect for `tiflash_fastscan` in both new session and global levels.
```

Comment (hongyunyan) [reply develop/dev-guide-use-fastscan.md:48]: You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect for the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` in session-level and global-level will both take the new value.

File: system-variables.md [hunk 2]
```diff
@@ -3067,6 +3067,13 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 - This variable is used to set the concurrency degree of the window operator.
 - A value of `-1` means that the value of `tidb_executor_concurrency` will be used instead.
 
+### `tiflash_fastscan` <span class="version-mark">New in v6.3.0</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: `false`
+- Range: `true | false`
+- If the FastScan feature is enabled (when set to `true`), TiFlash provides more efficient query performance, but does not guarantee the accuracy of the query results or data consistency.
```
Comment (hfxsd) [inline system-variables.md:3075]:
```suggestion
- If FastScan is enabled (set to `true`), TiFlash provides more efficient query performance, but does not guarantee the accuracy of the query results or data consistency.
```

File: develop/dev-guide-use-fastscan.md [hunk 3]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
+set session tiflash_fastscan=OFF;
+set global tiflash_fastscan=OFF;
 ```
 
 ## Mechanism of FastScan
 
 Data in the storage layer of TiFlash is stored in two layers: Delta layer and Stable layer.
 
-In Normal Mode, the TableScan operator processes data in the following steps:
+In normal mode (that is, when FastScan is not enabled), the TableScan operator processes data in the following steps:
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:65]:
```suggestion
In the default mode (that is, when FastScan is not enabled), the TableScan operator processes data in the following steps:
```

File: develop/dev-guide-use-fastscan.md [hunk 4]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
+The variable `tiflash_fastscan` supports session-level and global-level modifications. If you need to enable the FastScan feature in the current session, you can do so with the following statement:
+
+```
+set session tiflash_fastscan=ON;
 ```
 
-Once enabled, subsequent queries of this table in TiFlash will use the function of FastScan.
+You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect for the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` at session-level and global-level will both take the new value.
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:48]:
```suggestion
You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect for the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` of session level and global level will both take the new value.
```

File: develop/dev-guide-use-fastscan.md [hunk 5]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
++------------------+-------+
+| Variable_name    | Value |
++------------------+-------+
+| tiflash_fastscan | OFF   |
++------------------+-------+
+```
 
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
+The variable `tiflash_fastscan` supports session-level and global-level modifications. If you need to enable the FastScan feature in the current session, you can do so with the following statement:
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:42]:
```suggestion
You can configure the variable `tiflash_fastscan` at session level and global level. If you need to enable FastScan in the current session, you can do so with the following statement:
```

File: develop/dev-guide-use-fastscan.md [hunk 6]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
 
 By default, TiFlash guarantees the precision of query results and data consistency. With the feature FastScan, TiFlash provides more efficient query performance, but does not guarantee the accuracy of query results and data consistency.
 
-Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can enable FastScan for the corresponding table for querying.
+Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can enable the FastScan feature in the corresponding session. You can choose whether to enable the FastScan feature by modifying the value of the variable `tiflash_fastscan`.
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:16]:
```suggestion
Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can enable the FastScan feature in the corresponding session. You can choose whether to enable the FastScan feature by configuring the variable `tiflash_fastscan`.
```

File: develop/dev-guide-use-fastscan.md [hunk 7]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
 
-FastScan takes effect globally on tables that you enable FastScan by running [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md). TiFlash-related operations are not supported for temporary tables, in-memory tables, system tables, and tables with non-UTF-8 characters in column names.
+## Enable and disable FastScan
 
-For more information, see [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md).
+By default, the session and global level variables `tiflash_fastscan=OFF`, that is, the FastScan feature is not enabled. You can see the corresponding variable information by using the following statement.
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:20]:
```suggestion
By default, the session and global level variables `tiflash_fastscan=OFF`, that is, the FastScan feature is not enabled. You can view the corresponding variable information by using the following statement.
```

File: develop/dev-guide-use-fastscan.md [hunk 8]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
+The variable `tiflash_fastscan` supports session-level and global-level modifications. If you need to enable the FastScan feature in the current session, you can do so with the following statement:
+
+```
+set session tiflash_fastscan=ON;
 ```
 
-Once enabled, subsequent queries of this table in TiFlash will use the function of FastScan.
+You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect for the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` of session level and global level will both take the new value.
```
Comment (hfxsd) [inline develop/dev-guide-use-fastscan.md:48]:
```suggestion
You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect in the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` of session level and global level will both take the new value.
```

File: develop/dev-guide-use-fastscan.md [hunk 9]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
 
 By default, TiFlash guarantees the precision of query results and data consistency. With the feature FastScan, TiFlash provides more efficient query performance, but does not guarantee the accuracy of query results and data consistency.
 
-Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can enable FastScan for the corresponding table for querying.
+Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can enable the FastScan feature in the corresponding session. You can choose whether to enable the FastScan feature by configuring the variable `tiflash_fastscan`.
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:17]:
```suggestion
Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can enable the FastScan feature at the session or global level. You can choose whether to enable the FastScan feature by configuring the variable `tiflash_fastscan`.
```

File: develop/dev-guide-use-fastscan.md [hunk 10]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
 
-FastScan takes effect globally on tables that you enable FastScan by running [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md). TiFlash-related operations are not supported for temporary tables, in-memory tables, system tables, and tables with non-UTF-8 characters in column names.
+## Enable and disable FastScan
 
-For more information, see [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md).
+By default, the variable is `tiflash_fastscan=OFF` at the session level and global level , that is, the FastScan feature is not enabled. You can view the variable information by using the following statement.
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:21]:
```suggestion
By default, the variable is `tiflash_fastscan=OFF` at the session level and global level, that is, the FastScan feature is not enabled. You can view the variable information by using the following statement.
```

File: develop/dev-guide-use-fastscan.md [hunk 11]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
++------------------+-------+
+| Variable_name    | Value |
++------------------+-------+
+| tiflash_fastscan | OFF   |
++------------------+-------+
+```
 
-```sql
-ALTER TABLE table_name SET TIFLASH MODE FAST
+You can configure the variable `tiflash_fastscan` at session level and global level. If you need to enable FastScan in the current session, you can do so with the following statement:
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:43]:
```suggestion
You can configure the variable `tiflash_fastscan` at the session level and global level. If you need to enable FastScan in the current session, you can do so with the following statement:
```

File: develop/dev-guide-use-fastscan.md [hunk 12]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
+
+```
+set session tiflash_fastscan=ON;
 ```
 
-Once enabled, subsequent queries of this table in TiFlash will use the function of FastScan.
+You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect in the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` of session level and global level will both take the new value.
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:49]:
```suggestion
You can also set `tiflash_fastscan` at the global level. The new setting will take effect in new sessions, but will not take effect in the current and previous sessions. Besides, in new sessions, `tiflash_fastscan` of the session level and global level will both take the new value.
```

File: develop/dev-guide-use-fastscan.md [hunk 13]
```diff
@@ -13,39 +13,56 @@ This document describes how to use FastScan to speed up queries in Online Analyt
...
+set session tiflash_fastscan=OFF;
+set global tiflash_fastscan=OFF;
 ```
 
 ## Mechanism of FastScan
 
 Data in the storage layer of TiFlash is stored in two layers: Delta layer and Stable layer.
 
-In Normal Mode, the TableScan operator processes data in the following steps:
+In the default mode (that is, when FastScan is not enabled), the TableScan operator processes data in the following steps:
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:66]:
```suggestion
By default, FastScan is not enabled, and the TableScan operator processes data in the following steps:
```

File: develop/dev-guide-use-fastscan.md [hunk 14]
```diff
@@ -1,6 +1,7 @@
 ---
 title: FastScan
 summary: Introduces a way to speed up querying in OLAP scenarios by using FastScan.
+aliases: ['/docs/sql-statements/sql-statement-set-tiflash-mode/']
```
Comment (qiancai) [inline develop/dev-guide-use-fastscan.md:4]:
```suggestion
aliases: ['/tidb/dev/sql-statement-set-tiflash-mode']
```

File: system-variables.md [hunk 15]
```diff
@@ -3067,6 +3067,13 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 - This variable is used to set the concurrency degree of the window operator.
 - A value of `-1` means that the value of `tidb_executor_concurrency` will be used instead.
 
+### `tiflash_fastscan` <span class="version-mark">New in v6.3.0</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: `false`
+- Range: `true | false`
+- If FastScan is enabled (set to `true`), TiFlash provides more efficient query performance, but does not guarantee the accuracy of the query results or data consistency.
```
Comment (hfxsd) [inline system-variables.md:3075]:
```suggestion
- Scope: SESSION | GLOBAL
- Default value: `OFF`
- Range: `ON | OFF`
- If [FastScan](/develop/dev-guide-use-fastscan.md) is enabled (set to `ON`), TiFlash provides more efficient query performance, but does not guarantee the accuracy of the query results or data consistency.
```

**Review by qiancai — APPROVED** [review]
LGTM

File: develop/dev-guide-use-fastscan.md [hunk 16]
```diff
@@ -1,6 +1,7 @@
 ---
 title: FastScan
 summary: Introduces a way to speed up querying in OLAP scenarios by using FastScan.
+aliases: ['/tidb/dev/sql-statement-set-tiflash-mode']
```
Comment (joey-yez) [inline develop/dev-guide-use-fastscan.md:4]: 这个是必须的吗？

Comment (hfxsd) [reply develop/dev-guide-use-fastscan.md:4]: 是的，只要上过线的文档，即使删掉也会被搜索引擎搜出来，用户点开就会造成死链，所以需要加 aliases 避免死链。

**Review by joey-yez — APPROVED** [review]
LGTM

**Pull request merged**